service/history/api/getworkflowexecutionhistory/api.go

```diff
@@ -44,22 +44,23 @@ func Invoke(
 
 	isCloseEventOnly := request.Request.GetHistoryEventFilterType() == enumspb.HISTORY_EVENT_FILTER_TYPE_CLOSE_EVENT
 
-	// this function returns the following 7 things,
+	// this function returns the following 8 things:
 	// 1. the current branch token (to use to retrieve history events)
 	// 2. the workflow run ID
 	// 3. the last first event ID (the event ID of the last batch of events in the history)
 	// 4. the last first event transaction id
 	// 5. the next event ID
 	// 6. whether the workflow is running
-	// 7. error if any
-	queryHistory := func(
+	// 7. transient/speculative workflow task info
```

Comment:
😵 , could use a struct

Reply:
It is a func internal to another func. The entire `GetHistory` set of APIs need to be cleaned up and refactored. I don't want to make it more complex here.